src/regression_tracker.py

```diff
@@ -77,6 +66,28 @@ def _run(self, sub_id):
 {node['id']}")
                 self._create_regression(node, previous_nodes[0])
 
+    def _run(self, sub_id):
+        """Method to run regression tracking"""
+        self.log.info("Tracking regressions... ")
+        self.log.info("Press Ctrl-C to stop.")
+        sys.stdout.flush()
+
+        while True:
+            node = self._api_helper.receive_event_node(sub_id)
+            group_nodes = self._api.get_nodes({'parent': node['id']})
+
+            failures = []
+            for node in group_nodes:
+                if node['result'] == 'fail':
+                    failures.append(node)
+                failures.extend(self._api.get_nodes({
+                    'group': node['group'],
+                    'result': 'fail',
```

Comment:
I think this will also add the node in `group_nodes`, and the query should be more restrictive to pick nodes with the same parent path and same kernel revision to avoid other test results that have the same group name.  Or we could add a convention to have unique group names, like a group ID but I'm not sure if that's a good idea.

Reply:
This method is to get failed child nodes of the parent `checkout`.
I think you are referring to the logic for checking regression nodes. That is in  `_detect_regression` method.
So, first I am fetching all the child nodes with `fail` result recursively in `_run` and then the tracker will look for regression by iterating over each failed node in `_detect_regression`.

Reply:
Maybe I should add `path` filter while getting `previous_nodes`.